### PR TITLE
Table: Fix links for image cells

### LINKED ISSUE #31576: The combination of link and image is not working in table
I have a table that I want to use an image as a link
I added a label with a fix value and map it to a base64, it looks correct:
image

**What happened**:
I add a link to the cell, but it does not work
**What you expected to happen**:
Without the image mapping you can see that it is a link
image

**How to reproduce it (as minimally and precisely as possible)**:
Add a table, pick a value and map it to base64 image, add a link to that cell
**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.5
- Data source type & version: alertmanager
- OS Grafana is installed on: docker
- User OS & Browser: fedora
- Grafana plugins: alertmanager

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Adjust data link logic for table panel such that it is independent of the cell display override used.

**Which issue(s) this PR fixes**:
Closes #31576

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/TableCell.tsx [hunk 1]
```diff
@@ -32,15 +41,33 @@ export const TableCell: FC<Props> = ({ cell, field, tableStyles, onCellFilterAdd
...
-    <>
-      {cell.render('Cell', {
-        field,
-        tableStyles,
-        onCellFilterAdded,
-        cellProps,
-        innerWidth,
-      })}
-    </>
+  const link: LinkModel<any> | undefined = field.getLinks?.({
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Table/TableCell.tsx:44]: nit: no need for any
```suggestion
  const link: LinkModel | undefined = field.getLinks?.({
```

File: packages/grafana-ui/src/components/Table/TableCell.tsx [hunk 2]
```diff
@@ -32,15 +41,33 @@ export const TableCell: FC<Props> = ({ cell, field, tableStyles, onCellFilterAdd
...
+    valueRowIndex: rowIndex,
+  })[0];
+
+  let onClick: MouseEventHandler<HTMLAnchorElement> | undefined;
+  if (link && link.onClick) {
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Table/TableCell.tsx:49]: nit: can be simplified
```suggestion
  if (link?.onClick) {
```

**Review by hugohaggmark — APPROVED** [review]
Looks good and works as expected 🎉

**Review by dprokop — APPROVED** [review]
Works great!

**Comment by torkelo** [comment]
think different cells types needs to handle link logic & styling differently (think future cell types like graph cells, button cells, etc), not sure. With this change now the cell links have no text underline anymore. https://github.com/grafana/grafana/issues/33244

**Pull request merged**